### PR TITLE
Show helpful warning when @types/react isn't found

### PR DESCRIPTION
**Pull request by ijjk**
Since the default error shown when `@types/react` isn't installed is `JSX element implicitly has type 'any' because no interface 'JSX.IntrinsicElements' exists.` We show a separate warning when it is missing.

Currently, it just shows a warning instead of automatically installing it since it would be hard to know the correct way to install a dependency for a user considering they could be using `yarn`, `npm`, `pnpm`, etc.

### CODE REVIEW

File: packages/next/lib/verifyTypeScriptSetup.ts [hunk 1]
```diff
@@ -80,7 +80,7 @@ export async function verifyTypeScriptSetup(dir: string): Promise<void> {
         chalk.cyan.bold('typescript'),
         'by running',
         chalk.cyan.bold(
-          isYarn ? 'yarn add typescript' : 'npm install typescript',
+          isYarn ? 'yarn add typescript' : 'npm install --save-dev typescript',
```
Comment (resir014) [inline packages/next/lib/verifyTypeScriptSetup.ts]:
```suggestion
          isYarn ? 'yarn add --dev typescript' : 'npm install --save-dev typescript',
```

We need the one for yarn to be saved in `devDependencies` as well 😅

Comment (ijjk) [reply packages/next/lib/verifyTypeScriptSetup.ts]: Good catch

**Review by lfades — APPROVED** [review]

**Review by timneutkens — COMMENTED** [review]
Could we collect these checks so that users only see 1 command to be ran, eg typescript + @types/react etc.

**Comment by ijjk** [comment]
Yup, it's updated to show them together if they both need to be installed now.

**Review by timneutkens — APPROVED** [review]

**Pull request merged**